modules/living_spaces_group/living_spaces_group.module

```diff
@@ -1105,3 +1105,25 @@ function living_spaces_group_group_update(GroupInterface $group) {
   }
 
 }
+
+/**
+ * Implements hook_query_TAG_alter().
+ */
+function living_spaces_group_query_living_spaces_menus_user_spaces_alter(Drupal\Core\Database\Query\AlterableInterface $query) {
+  if ($spaces = \Drupal::config('living_spaces_group.exclude_spaces')->get('spaces')) {
+    $excluded = array_intersect($spaces, ['mail_counseling', 'on_site_counseling', 'telephone_counseling', 'chat_counseling']);
+
+    if ($excluded) {
+      $query->condition('groups_field_data.type', array_values($excluded), 'NOT IN');
+    }
+  }
+
+  $query->leftJoin('user__joined_spaces', 'user__joined_spaces', 'user__joined_spaces.joined_spaces_target_id = groups_field_data.id');
+  $query->condition('user__joined_spaces.entity_id', \Drupal::currentUser()->id(), '=');
+
+  $query->leftJoin('user__field_favorite_spaces', 'user__field_favorite_spaces', 'user__field_favorite_spaces.field_favorite_spaces_target_id = groups_field_data.id AND user__field_favorite_spaces.entity_id = user__joined_spaces.entity_id');
+  $query->fields('user__field_favorite_spaces', ['delta', 'entity_id']);
+
+  $order_by = &$query->getOrderBy();
+  $order_by = array_merge(['user__field_favorite_spaces.delta' => 'DESC'], $order_by);
```

Comment:
Shouldn't it be ordered ascending and not descending? Right now it will be showing favorites in opposite order than it is on entity edit

Reply:
Done! I've added sorting to the code because $query->addExpression with IFNULL doesn't work properly.